app/models/user.rb

```diff
@@ -653,6 +653,14 @@ def publish_do_not_disturb(ends_at: nil)
     MessageBus.publish("/do-not-disturb/#{id}", { ends_at: ends_at&.httpdate }, user_ids: [id])
   end
 
+  def publish_user_status(status)
+    payload = status ?
+                { description: status.description, emoji: status.emoji } :
+                nil
+
+    MessageBus.publish("/user-status/#{id}", payload, user_ids: [id])
```

Comment:
this is going to be tricky security wise.

this is only making it public to the actual user. However in scenarios like chat we want to make "Status" public to a far larger group. I guess we start from the fundamental, "who can see status" ? If it is everyone then this should be published on a different channel ... eg : `/user-status` with no postfix.

Reply:
That makes sense. I think that's something we'll consider carefully when we start showing status in more places, including chat (that will be soon). Deciding how to propagate statuses of other users is the trickiest part of showing status on the chat. 

For now, making it public to the actual user it's all we need, since we only show status:
- on user card (and we don't need Message Bus there) 
- to the actual user on the user menu. 

So, now, we send this message only to update the status of the actual user in all opened tabs of their browsers.